### DISCUSSION TITLE
Possible to run SDXL via Colab Pro (not Pro+)?

### DISCUSSION THREAD

**Discussion by WorldofDepth** [TheLastBen/fast-stable-diffusion#2412] [General]
Hi, does anyone know if this is possible, perhaps by using the --medvram / --lowvram options? I don't know where to insert those in the notebook to try them out (tried to put them after instances of webui.py in the last code block, but that did not work).

Colab Pro gives you 12.7GB of system RAM / GPU RAM, and loading SDXL in the latest notebook doesn't appear to use more than 7GB RAM at any moment, but it gives ^C before creating the model / loading weights finishes.

If this is not possible, what are people's recommendations for alternatives, when running locally is not an option? Thanks for any help!

**Comment by TheLastBen** [comment]
set the runtime shape to High-RAM, in the menu "runtime"

**Reply by WorldofDepth** [reply]
I did check that I was already in high RAM mode. Is there anything else to try—the --lowvram option or other? Thank you!

**Reply by TheLastBen** [reply]
set it to high-ram, and post a screenshot of the cell's output

**Reply by WorldofDepth** [reply]
Thank you for the help! Here are runtime details and the last cell output:

<img width="297" alt="Screen Shot 2023-08-06 at 10 52 29 PM" src="https://github.com/TheLastBen/fast-stable-diffusion/assets/77465052/8dfcbd33-bc3c-4446-96db-cbca783d13ff">

<img width="766" alt="Screen Shot 2023-08-06 at 10 52 39 PM" src="https://github.com/TheLastBen/fast-stable-diffusion/assets/77465052/df73c6cf-2aee-4d50-a9a1-a16701252644">

I can paste the entire text here if it helps. The memory error line reads:
`torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 14.00 MiB (GPU 0; 14.75 GiB total capacity; 12.77 GiB already allocated; 12.81 MiB free; 13.41 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`

**Comment by phephanhoang** [comment]
Same. Set High-Ram can't solve the problem.

**Comment by TheLastBen** [comment]
test run it without the first cell to make sure the issue isn't caused by some extension

**Reply by WorldofDepth** [reply]
Do you mean the "Connect Google Drive" cell? I do not use that one.

**Reply by TheLastBen** [reply]
you're running the notebook without gdrive ?

**Reply by WorldofDepth** [reply]
Yes, correct

**Reply by TheLastBen** [reply]
try adding `--medvram`

**Reply by WorldofDepth** [reply]
OK—where exactly should that be added? I had trouble with that before. Thank you!

**Reply by ClaraSanders** [reply]
Any Colab Pro users know if you can run XL with only High Ram selected? I mean on T4, after the computing units are used.
